lib/puppet/provider/exec/posix.rb

```diff
@@ -39,6 +43,8 @@ def checkexe(command)
   end
 
   def run(command, check = false)
+   command = command.split if command.respond_to?(:split) && !resource[:pass_through_shell]
```

Comment:
Unfortunately this won't work if the the `command` contains single or double quotes, e.g. `/usr/bin/a "b c"`. Can we apply a regex similar to  https://github.com/ruby/ruby/blob/e2599a358d6b6a83f2da0972e1f91eccbf69c55e/lib/shellwords.rb#L91, but skip the part where quotes and special characters are escaped?

Reply:
Added the regex but not sure if it covers all the use-case.